Core/tests/DummyPlatform.cpp

```diff
@@ -396,14 +463,53 @@ void SetupProjectWithDummyPlatform(gd::Project& project,
         gd::make_unique<gd::BehaviorsSharedData>());
   }
 
+  {
+    gd::BehaviorMetadata &effectBehavior =
+        extension
+            ->AddBehavior("EffectBehavior",
+                          _("Effect capability"),
+                          "Effect",
+                          _("Apply visual effects to objects."),
+                          "",
+                          "res/actions/effect24.png", "EffectBehavior",
+                          std::make_shared<gd::Behavior>(),
+                          std::make_shared<gd::BehaviorsSharedData>())
+            .SetHidden();
+
+    // Add this expression for the effect capability.
+    effectBehavior
+        .AddStrExpression("GetSomethingRequiringEffectCapability",
+                          "Get something, but this requires the effect "
+                          "capability for the object.",
+                          "",
+                          "",
+                          "")
+        .AddParameter("object", _("Object"), "")
+        .AddParameter("behavior", _("Behavior"), "EffectBehavior")
+        .AddParameter("expression", _("Number parameter"))
+        .SetFunctionName("getSomethingRequiringEffectCapability");
+  }
   {
     auto& object = extension
                        ->AddObject<gd::ObjectConfiguration>(
-                           "FakeObjectWithUnsupportedCapability",
-                           "FakeObjectWithUnsupportedCapability",
-                           "This is FakeObjectWithUnsupportedCapability",
+                           "FakeObjectWithDefaultBehavior",
+                           "FakeObjectWithDefaultBehavior",
+                           "This is FakeObjectWithDefaultBehavior",
                            "")
-                       .AddUnsupportedBaseObjectCapability("effect");
+                       .AddDefaultBehavior("MyExtension::EffectBehavior");
```

Comment:
An object with the effect capability VS a fake Sprite that doesn't have it is against our functional model logic. I didn't add the capability on Sprite because a lot of tests are using this object and it would require many changes.
But it's an important technical dept as it's misleading on how Sprite objects work.

Do you think I should add a TODO or make the changes in this PR?

Reply:
Let's add a to-do. 

Someone using this test data can understand that it's not modèles 100% on reality :)